internal/powerbi/resource_pbix.go

```diff
@@ -303,11 +304,16 @@ func setPBIXParameters(d *schema.ResourceData, meta interface{}) error {
 
 	client := meta.(*powerbiapi.Client)
 	parameter := d.Get("parameter").(*schema.Set)
-	datasetID := d.Get("dataset_id").(string)
+	datasetID, datasetOk := d.GetOk("dataset_id")
 	groupID := d.Get("workspace_id").(string)
 	if parameter != nil {
 		parameterList := parameter.List()
 		if len(parameterList) > 0 {
+
+			if !datasetOk {
+				return fmt.Errorf("Unable to update parameters on a PBIX file that does not contain a dataset")
```

Comment:
Not sure if this should really be considered an "error", as technically it is completely fine to have pbix resources without datasets.

Reply:
That code is for when updating parameters. This error occurs in the case if you have a PBIX without a dataset and try to set parameters (or datasources, which has similar code) in the terraform

```
resource "powerbi_workspace" "test" {
	name = "My Worksapce"
}

resource "powerbi_pbix" "test" {
	workspace_id = "${powerbi_workspace.test.id}"
	name = "PBIX without a dataset"
	source = "./pbix_wihtout_dataset.pbix"
	source_hash = "filemd5(./pbix_wihtout_dataset.pbix)"

# Setting a parameter - but as parameters are set at the dataset this doesnt make any sense if the pbix doesnt have a dataset
	parameter {
		name = "ParamOne"
		value = "NewParamValueOne"
	}
}
```

I could ignore the parameters for cases of datasetless pbix, but if you have this case something is probably happening that you are not expecting, so think its safer to throw an error

Reply:
Makes sense!